app/cdap/components/WrangleHome/index.tsx

```diff
@@ -17,16 +17,22 @@
 import { Typography } from '@material-ui/core';
 import Box from '@material-ui/core/Box';
 import LoadingSVG from 'components/shared/LoadingSVG';
+import OngoingDataExploration from 'components/WrangleHome/Components/OngoingDataExploration/index';
+import WrangleCard from 'components/WrangleHome/Components/WrangleCard/index';
+import WrangleHomeTitle from 'components/WrangleHome/Components/WrangleHomeTitle/index';
+import { GradientLine, HeaderImage } from 'components/WrangleHome/icons';
+import { useStyles } from 'components/WrangleHome/styles';
+import T from 'i18n-react';
 import React, { useState } from 'react';
-import OngoingDataExploration from './Components/OngoingDataExploration';
-import WrangleCard from './Components/WrangleCard';
-import WrangleHomeTitle from './Components/WrangleHomeTitle';
-import { GradientLine, HeaderImage } from './icons';
-import { useStyles } from './styles';
+import { Link } from 'react-router-dom';
+import { getCurrentNamespace } from 'services/NamespaceStore';
 
-export default function WranglerHome() {
+export default function() {
   const classes = useStyles();
   const [loading, setLoading] = useState(false);
+  localStorage.setItem('addConnectionRequestFromNewUI', 'home');
+
+  const [viewAllLink, toggleViewAllLink] = useState<boolean>(false);
```

Comment:
This shouldn't be in state you should be able to just have the view all link with a {connectorTypes.length > 5  && <ViewAllLink />} or something like that

Reply:
Actually, connectorTypes data is getting fetched in a util function which is being called in child (WrangleCard) component of the parent component(WrangleHome). So, just to get a conclusion to show the view all link will be decided in that function, which is why we are passing this state updating function (toggleViewAllLink ) to the child component.